buildlib/azure-pipelines.yml

```diff
@@ -10,33 +10,43 @@ resources:
     - container: centos7
       image: ucfconsort.azurecr.io/ucx/centos7:1
       endpoint: ucfconsort_registry
+    - container: fedora
+      image: ucfconsort.azurecr.io/ucx/fedora:30
+      endpoint: ucfconsort_registry
 
 stages:
-  - stage: Build
+  - stage: LatestCCs
     jobs:
-      - job: Compile
-        displayName: Compile Tests
-        pool:
-          vmImage: 'Ubuntu-16.04'
+      - job: latest_cc
+        displayName: Latest CCs and CppCheck
+        container: fedora
         steps:
-          - bash: |
-              ./autogen.sh
+          - bash: ./autogen.sh
             displayName: Setup autotools
 
-          - bash: |
-              mkdir build && cd build
-              ../configure --disable-numa
+          - bash: ./contrib/configure-release
```

Comment:
You need to force -Werror or AZP will not give a red X if new warnings are introduced.

Reply:
Also, I don't think it is a good idea to do test builds with debug code disabled. This can lead to missing failures in optional debug code, and bogus warnings like 'variable not used' when assertions are stripped out. Not desirable for CI.

Reply:
we want to pass static checks with release builds as well, because redhat are running their checks on a release config

Reply:
We have `-Werror` always enabled: https://github.com/openucx/ucx/blob/080c231d3c9861538738d64e8f71283f9a102866/config/m4/compiler.m4#L11

Reply:
You should not enable -Werror by default like that, old distros have crazy compilers and it is very hard to maintain warning free code in that environment.

I think you should focus this part of the pipeline on getting coverage and removing as many static defects as possible. Worry about release build/etc in other stages.

Reply:
That is not what I change in this PR.